File: Tecnicell.Server/Models/Responses/AccessoryResponse.cs
using Tecnicell.Server.Models.Entity;
using Tecnicell.Server.Models.ViewModel.Phone;
using Tecnicell.Server.Models.ViewModel;
using Tecnicell.Server.Models.ViewModel.Battery;
using Tecnicell.Server.Models.ViewModel.Accessory;

namespace Tecnicell.Server.Models.Responses
{
    public class AccessoryResponse
    {
        public AccessoryView? View { get; set; }
        public ImageViewModel? Image { get; set; }
        public IEnumerable<AccessoryHistoryViewModel> Histories { get; set; } = new List<AccessoryHistoryViewModel>();
    }
}

File: tecnicell.client/src/app/Interfaces/business/ApiResponses/AccessoryResponse.ts
import { AccessoryHistory } from "../Models/Accessory";
import { Image } from "../Models/Image";

export interface AccessoryResponse {
    view:      AccessoryView;
    image:     Image;
    histories: AccessoryHistory[];
}
export interface AccessoryView {
    code:      string;
    type:      string;
    typeCode: string;
    name:      string;
    salePrice: number;
    imageCode: string;
    quantity:  number;
    available: string;
}
